src/cryptonotecore/LevelDBWrapper.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2012-2017, The CryptoNote developers, The Bytecoin developers
+// Copyright (c) 2018-2019, The TurtleCoin Developers
+//
+// Please see the included LICENSE file for more information.
+
+#include "LevelDBWrapper.h"
+
+#include "DataBaseErrors.h"
+#include "leveldb/cache.h"
+#include "leveldb/db.h"
+#include "leveldb/table.h"
+#include "leveldb/write_batch.h"
+
+using namespace CryptoNote;
+using namespace Logging;
+
+namespace
+{
+    const std::string DB_NAME = "LevelDB";
+}
+
+LevelDBWrapper::LevelDBWrapper(std::shared_ptr<Logging::ILogger> logger):
+    logger(logger, "LevelDBWrapper"),
+    state(NOT_INITIALIZED)
+{
+}
+
+LevelDBWrapper::~LevelDBWrapper() {}
+
+void LevelDBWrapper::init(const DataBaseConfig &config)
+{
+    // Set up database connection information and open database
+    leveldb::DB* dbPtr;
+
+    leveldb::Options dbOptions;
+#if defined(USE_LEVELDB)
+    logger(INFO) << "Daemon uses kSnappyCompression...";
+    dbOptions.compression = leveldb::kSnappyCompression;
+#else
+    logger(INFO) << "Daemon uses kNoCompression...";
+    dbOptions.compression = leveldb::kNoCompression;
+#endif
+    dbOptions.max_file_size = 1024 * 1024 * 1024; // 1024 MB
+
+    dbOptions.write_buffer_size =  static_cast<size_t>(config.getWriteBufferSize());
+
+    dbOptions.max_open_files =  config.getMaxOpenFiles();
+
+    dbOptions.block_cache =  leveldb::NewLRUCache(config.getReadCacheSize());
+
+    if (state.load() != NOT_INITIALIZED)
+    {
+        throw std::system_error(make_error_code(CryptoNote::error::DataBaseErrorCodes::ALREADY_INITIALIZED));
+    }
+
+    std::string dataDir = getDataDir(config);
+
+    leveldb::Status status = leveldb::DB::Open(dbOptions, dataDir, &dbPtr);
+
+    logger(INFO) << "Opening DB in " << dataDir;
+
+    if (true == status.ok())
```

Comment:
No need to compare to true.
```suggestion
    if (status.ok())
```